src/backend/xdg_desktop_portal.rs

```diff
@@ -0,0 +1,284 @@
+use std::path::PathBuf;
+
+use crate::backend::DialogFutureType;
+use crate::file_dialog::Filter;
+use crate::{FileDialog, FileHandle};
+
+use ashpd::desktop::file_chooser::{
+    FileChooserProxy, FileFilter, OpenFileOptions, SaveFileOptions,
+};
+// TODO: convert raw_window_handle::RawWindowHandle to ashpd::WindowIdentifier
```

Comment:
Maybe we should provide a feature for that in ashpd

Reply:
That would be nice.

Reply:
https://github.com/bilelmoussaoui/ashpd/issues/40